evaluation_script/main.py

```diff
@@ -1,5 +1,48 @@
-import os
-import gzip
+from .utils import get_file_extension
+from .utils import open_file
+from .utils import are_all_base_metrics_zero
+from .utils import calculate_precision
+from .utils import calculate_recall
+from .utils import calculate_fpr
+from .utils import calculate_f1
+from .utils import BaseMetrics
+from .utils import Metrics
+
+query_level_base_metrics = {}
+query_level_metrics = {}
+documents_with_ground_truth = set()
+
+"""
+    This function calculates query level metrics from query level base metrics
+"""
+def calculate_query_level_metrics():
+    qa_precision = 0
+    qa_recall = 0
+    qa_fpr = 0
+    qa_f1 = 0
+
+    for query_id in query_level_base_metrics:
+        base_metrics = query_level_base_metrics[query_id]
+        if not are_all_base_metrics_zero(base_metrics.tp, base_metrics.tn, base_metrics.fp, base_metrics.fn):
+            query_level_metrics[query_id].precision = calculate_precision(base_metrics.tp, base_metrics.fp)
+            query_level_metrics[query_id].recall = calculate_recall(base_metrics.tp, base_metrics.fn)
+            query_level_metrics[query_id].fpr = calculate_fpr(base_metrics.fp, base_metrics.tn)
+            query_level_metrics[query_id].f1 = calculate_f1(base_metrics.tp, base_metrics.fp, base_metrics.fn)
+
+    for query_id in query_level_base_metrics:
+        qa_precision = qa_precision + query_level_metrics[query_id].precision
+        qa_recall = qa_recall + query_level_metrics[query_id].recall
+        qa_fpr = qa_fpr + query_level_metrics[query_id].fpr
+        qa_f1 = qa_f1 + query_level_metrics[query_id].f1
+
+    total_queries = len(query_level_base_metrics.keys())
+    if total_queries > 0:
```

Comment:
`total_queries > 0` is handling an unexpected edge case. Would be clearer to put it at the start of processing, right after the declarations of `qa_xzy` variables.

Reply:
Will fix this.

Reply:
Fixed in revision